syncapi/sync/requestpool.go

```diff
@@ -407,6 +407,14 @@ func (rp *RequestPool) OnIncomingSyncRequest(req *http.Request, device *userapi.
 						)
 					},
 				),
+				MultiRoomDataPosition: withTransaction(
+					syncReq.Since.MultiRoomDataPosition,
+					func(snapshot storage.DatabaseTransaction) types.StreamPosition {
```

Comment:
Are we using the `snapshot` terminology for `storage.DatabaseTransaction` instances deferring from the original `txn` on purpose? If not, maybe we should maintain the consistency with upstream terminology?

Reply:
They use `txn`, but later they call the same parameter a `snapshot`. I have renamed it to `txn` to keep consistency at least in this file.